### PR TITLE
Add recall per-sample plot script

### PR DESCRIPTION
**Pull request by mbhall88**
addresses #2 

Also fixed an issue where the path to the precision csv files was not pointing at the filtered ones.

### CODE REVIEW

**Comment by mbhall88** [comment]
recall_per_sample

File: scripts/csv_to_paper_plots/precision_per_sample/precision_per_sample.py [hunk 1]
```diff
@@ -5,12 +5,12 @@
 
 plt.style.use("ggplot")
 
-csvfile = "../pandora1_paper_analysis_output_20_way/illumina_analysis/precision_per_ref_per_clade/precision_per_ref_per_clade_samtools_pandora.csv"
+csvfile = "../pandora1_paper_analysis_output_20_way/illumina_analysis_with_filters/precision_per_ref_per_clade/precision_per_ref_per_clade_samtools_pandora.csv"
```
Comment (leoisl) [inline scripts/csv_to_paper_plots/precision_per_sample/precision_per_sample.py:8]: I think we should keep looking at the non-filtered results, filtered results will go in supp

File: scripts/csv_to_paper_plots/recall_per_sample/recall_per_sample.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+from matplotlib import cm
+
+plt.style.use("ggplot")
+
+csvfile = "recall_per_ref_per_clade_samtools_pandora.csv"
```
Comment (leoisl) [inline scripts/csv_to_paper_plots/recall_per_sample/recall_per_sample.py:8]: Could we get the csvfile directly from the package, like in the precision script? We can also remove all the .csv that were added, just the script will be then fine

**Review by leoisl — CHANGES_REQUESTED** [review]

**Comment by leoisl** [comment]
Could we also fix the colour of the tools so that they follow the pattern in both the top figures of https://user-images.githubusercontent.com/8397967/95141283-9f90eb00-0768-11eb-8dd7-1c598d5a3d0e.png ?

**Comment by iqbal-lab** [comment]
Important to see filtered too! Was surprised by this, but it is TAR, which is not going into the paper as the counting is confusing

File: scripts/csv_to_paper_plots/precision_per_sample/precision_per_sample.py (hunk 1 above)
Comment (mbhall88) [reply scripts/csv_to_paper_plots/precision_per_sample/precision_per_sample.py:8]: Seriously? Why would we look at unfiltered for? I'm confused by this statement in conjunction with https://github.com/leoisl/pandora1_paper_full_pipeline/pull/6#issuecomment-704022030

File: scripts/csv_to_paper_plots/recall_per_sample/recall_per_sample.py (hunk 2 above)
Comment (mbhall88) [reply scripts/csv_to_paper_plots/recall_per_sample/recall_per_sample.py:8]: There's no package in this repo though?

**Comment by leoisl** [comment]
> Important to see filtered too! Was surprised by this, but it is TAR, which is not going into the paper as the counting is confusing

I am confused by this. 6c and d are precision per sample for illumina and nanopore. Figures 8a and 8b are the per clade plots. These 4 plots are based on TAR restricted to a given sample, as TAR is the only recall measure defined for a single sample. We can rephrase TAR for a single sample `S` as the number of alleles found over the number of alleles in all pan vars that are in `S`.

**Comment by iqbal-lab** [comment]
OK, i agree what you are saying is true.  This is i think the first time i've seen this plot with the filtered data. oh no - it's actually the first time at all isnt it. can we see with/out filtering please?

**Comment by iqbal-lab** [comment]
v interesting, but it recapitulates what we have in the multi-panel histogram, so will not be in the paper

**Comment by leoisl** [comment]
just to be sure, the counterpart, precision per sample remains as Figure 6c and 6d?

**Comment by iqbal-lab** [comment]
yes

File: scripts/csv_to_paper_plots/precision_per_sample/precision_per_sample.py (hunk 1 above)
Comment (leoisl) [reply scripts/csv_to_paper_plots/precision_per_sample/precision_per_sample.py:8]: From what I can read from the text, the filtered plots are going into supplementary. In the end, we will need both, unfiltered ones going into the main text. I could be wrong though...

File: scripts/csv_to_paper_plots/recall_per_sample/recall_per_sample.py (hunk 2 above)
Comment (leoisl) [reply scripts/csv_to_paper_plots/recall_per_sample/recall_per_sample.py:8]: The packages are the 20- and 4-way packages, that are automatically downloaded here: https://github.com/leoisl/pandora1_paper_full_pipeline/blob/master/scripts/csv_to_paper_plots/generate_paper_plots.sh (sorry, I was sure I mentioned this somewhere...)

**Comment by leoisl** [comment]
Zam prefers all plots in the paper in `R`, so will not merge this PR, but will keep it open, as I will add it into the pipeline later

**Pull request closed without merging**